src/main/scala/is/hail/linalg/BlockMatrix.scala

```diff
@@ -295,6 +379,14 @@ class BlockMatrix(val blocks: RDD[((Int, Int), BDM[Double])],
     new BlockMatrix(zeroedBlocks, blockSize, nRows, nCols)
   }
   
+  def filterRectangles(flattenedRectangles: Array[Long]): BlockMatrix = {
+    require(flattenedRectangles.length % 4 == 0)
+    (1 until flattenedRectangles.length by 2).foreach(flattenedRectangles(_) -= 1) // make stops inclusive
```

Comment:
I think this logic is way more confusing than it needs to be. If this is the function signature called from Python, maybe the first step should be to convert to an array of case classes?

Reply:
So this and the next comment are correctly point out confusing code due to a difference in the python interface where I made rectangles are left-inclusive, right-exclusive, and the earlier Scala notion that was left- and right-inclusive. I've rewritten the Scala to be right-exclusive as well, which removes these bits while simplifying some other logic. I don't think the case class is necessary now, but am open to making the change.